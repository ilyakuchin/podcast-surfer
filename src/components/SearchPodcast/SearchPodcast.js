import React, { Component } from "react";
import findPodcasts from "../../api/FindPodcasts";
import { Link } from "react-router-dom";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faSearch } from "@fortawesome/free-solid-svg-icons";

export default class SearchPodcast extends Component {
  constructor(props) {
    super(props);

    this.state = {
      podcasts: [],
      searchPhrase: ""
    };

    this.handleChange = this.handleChange.bind(this);
  }
  render() {
    return (
      <div>
        <form>
          <input
            type="text"
            placeholder="Search.."
            name="search"
            onChange={this.handleChange}
          />
          <button
            type="submit"
            onClick={e => {
              e.preventDefault();

              findPodcasts(this.state.searchPhrase).then(res =>
                this.setState({ podcasts: res })
              );
            }}
          >
            <FontAwesomeIcon icon={faSearch} />
          </button>
        </form>

        <ul>
          {this.state.podcasts.map(item => (
            <li key={item.id}>
              <Link to="/podcast">{item.name}</Link>
              <img src={item.image} alt="podcast cover" />
            </li>
          ))}
        </ul>
      </div>
    );
  }

  handleChange(e) {
    this.setState({ searchPhrase: e.target.value });
  }
}
