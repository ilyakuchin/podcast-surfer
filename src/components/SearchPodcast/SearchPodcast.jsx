import React from 'react';
import styled from 'styled-components';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import ConnectedSearchForm from '../SearchForm/SearchForm';
import ConnectedSearchResults from '../SearchResults/SearchResults';
import PopularPodcasts from '../PopularPodcasts/PopularPodcasts';

const ComponentGrid = styled.div`
  display: grid;
  height: 100%;
  grid-template-rows: 50px auto;
  justify-items: center;
  grid-template-areas:
    'search'
    'results';
`;

export function SearchPodcast({ podcasts }) {
  function hasPodcasts() {
    return podcasts.length !== 0;
  }
  return (
    <ComponentGrid>
      <ConnectedSearchForm />
      {!hasPodcasts() ? <PopularPodcasts /> : <ConnectedSearchResults />}
    </ComponentGrid>
  );
}

SearchPodcast.propTypes = {
  podcasts: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string,
      image: PropTypes.string,
      rss: PropTypes.string,
      name: PropTypes.string
    })
  ).isRequired
};

const mapStateToProps = state => {
  return {
    podcasts: state.podcasts.podcasts
  };
};

const ConnectedSearchPodcast = connect(mapStateToProps)(SearchPodcast);

export default ConnectedSearchPodcast;
