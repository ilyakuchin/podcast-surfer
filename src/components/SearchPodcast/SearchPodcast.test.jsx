import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import SearchPodcast from './SearchPodcast';

configure({ adapter: new Adapter() });

describe('Test SearchPodcast Component', () => {
  test('should render correctly', () => {
    expect(toJson(shallow(<SearchPodcast />))).toMatchSnapshot();
  });
});
