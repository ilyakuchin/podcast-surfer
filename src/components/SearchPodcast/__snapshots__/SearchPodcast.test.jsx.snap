// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SearchPodcast Component should render correctly 1`] = `
<styled.div>
  <Connect(SearchForm) />
  <Connect(PopularPodcasts) />
</styled.div>
`;

exports[`Test SearchPodcast Component should show popular podcasts when has no podcasts in search results 1`] = `
<styled.div>
  <Connect(SearchForm) />
  <Connect(SearchResults) />
</styled.div>
`;
